kythe/typescript/indexer.ts

```diff
@@ -682,24 +682,38 @@ class Visitor {
     this.emitEdge(this.newAnchor(decl.name), 'defines/binding', kType);
 
     if (decl.typeParameters) this.visitTypeParameters(decl.typeParameters);
-    this.visitType(decl.type);
-    // TODO: in principle right here we emit an "aliases" edge.
-    // However, it's complicated by the fact that some types don't have
-    // specific names to alias, e.g.
-    //   type foo = number|string;
-    // Just punt for now.
+    const kAlias = this.visitType(decl.type);
+    // Emit an "aliases" edge if the aliased type has a singular VName.
+    if (kAlias) {
+      this.emitEdge(kType, 'aliases', kAlias);
+    }
   }
 
   /**
    * visitType is the main dispatch for visiting type nodes.
    * It's separate from visit() because bare ts.Identifiers within a normal
    * expression are values (handled by visit) but bare ts.Identifiers within
    * a type are types (handled here).
+   *
    * @return the VName of the type, if available.  (For more complex types,
    *    e.g. Array<string>, we might not have a VName for the specific type.)
    */
   visitType(node: ts.Node): VName|undefined {
     switch (node.kind) {
+      case ts.SyntaxKind.TypeReference:
+        const ref = node as ts.TypeReferenceNode;
+        const kType = this.visitType((node as ts.TypeReferenceNode).typeName);
```

Comment:
I think calling visitType on the typeName here will cause emits of the type without the generic, is that what we want?  Or should this bail entirely when there are type arguments?

I guess when you see "let x: Array<string>" we still want "Array" to be clickable and to go to the defn of array, without generics, so I think I convined myself that this is correct as written.

Reply:
yeah it'll emit a ref for `Array` in `Array<string>` but bail on emitting anything for `Array<string>` as a whole.